src/FixedDataTable.js

```diff
@@ -428,28 +424,67 @@ var FixedDataTable = React.createClass({
     }
 
     // In the case of controlled scrolling, notify.
-    if (this.props.ownerHeight !== nextProps.ownerHeight ||
-        this.props.scrollTop !== nextProps.scrollTop ||
+    if (ownerHeight !== nextProps.ownerHeight ||
+        scrollTop !== nextProps.scrollTop ||
         nextProps.scrollLeft !== undefined) {
       this._didScrollStart();
     }
     this._didScrollStop();
-
-    this.setState(this._calculateState(nextProps, this.state));
   },
 
   componentDidUpdate() {
     this._reportContentHeight();
   },
 
   render() /*object*/ {
-    var state = this.state;
-    var props = this.props;
+    const {
+      fixedColumnGroups,
+      fixedColumns,
+      scrollableColumnGroups,
+      scrollableColumns,
+    } = columnDetailsSelector(this.props);
+
+    const {
+      bodyHeight,
+      componentHeight,
+      footerHeight,
```

Comment:
I could technically get footer and headerHeight from props but getting from here for consistency